File: internal/models/circuit.go
package models

import "time"

type Circuit struct {
	ID          int64     `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Data        string    `json:"data"` // JSON string containing circuit diagram data
	UserID      int64     `json:"user_id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type Component struct {
	ID       int64  `json:"id"`
	Type     string `json:"type"`     // resistor, capacitor, etc.
	Name     string `json:"name"`
	Value    string `json:"value"`
	Standard string `json:"standard"` // KEC, IEC, UL
}

File: src/ElectricCircuitWeb.API/Models/Circuit.cs
namespace ElectricCircuitWeb.API.Models;

/// <summary>
/// 회로 모델
/// </summary>
public class Circuit
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string ProjectId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Data { get; set; } = string.Empty; // JSON으로 저장되는 회로 데이터
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}
